### PR TITLE
Add ttnn argmax op

### PR DESCRIPTION
**Pull request by TT-BrianLiu**
- Port over ttlib argmax op, program, and kernels
- Remove risc_v op folder from ttlib
- Add new ttnn/cpp/ttnn/op_library/reduction op folder to ttnn
- Remove unused python ttnn reduce op and validate from ttnn/ttnn/operations/reduction.py

### CODE REVIEW

File: ttnn/cpp/pybind11/operations/reduction.hpp [hunk 1]
```diff
@@ -36,12 +36,41 @@ void bind_reduction_operation(py::module& module, const reduction_operation_t& o
...
+    // Generic reductions
     detail::bind_reduction_operation(module, ttnn::sum);
     detail::bind_reduction_operation(module, ttnn::mean);
     detail::bind_reduction_operation(module, ttnn::max);
     detail::bind_reduction_operation(module, ttnn::min);
     detail::bind_reduction_operation(module, ttnn::std);
     detail::bind_reduction_operation(module, ttnn::var);
+
+    // Special reductions
+    ttnn::bind_registered_operation(
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/pybind11/operations/reduction.hpp:48]: please lets move this chunk to a separate method

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/reduction.hpp [hunk 2]
```diff
@@ -163,9 +164,28 @@ struct Reduce {
...
+        return {ttnn::TensorSchema{
+            4, 4, {ttnn::bfloat16}, {ttnn::ROW_MAJOR_LAYOUT}, true, false, false, false}};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const Tensor& input_tensor, Args&&... args) {
+        return std::forward_as_tuple(input_tensor);
+    }
+
+    static ttnn::Tensor execute_on_worker_thread(
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/reduction.hpp:178]: I recommend to pass `optional output_tensor` and another overload with `queue_id` in this PR. 
operation::run supports those arguments already so it is just a few lines.

**Review by xanderchin — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/reduction.hpp (hunk 2 above)
Comment (TT-BrianLiu) [reply ttnn/cpp/ttnn/operations/reduction.hpp:178]: will do soon

**Comment by TT-BrianLiu** [comment]
Passing pipelines:
- post-commit all: https://github.com/tenstorrent/tt-metal/actions/runs/9423130911
- post-commit ttnn unit tests: https://github.com/tenstorrent/tt-metal/actions/runs/9423131946

**Review by tt-aho — APPROVED** [review]
lgtm

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**